src/main/java/gregtech/api/metatileentity/MetaTileEntityHolder.java

```diff
@@ -83,6 +95,7 @@ public void notifyBlockUpdate() {
     @Override
     public void readFromNBT(@Nonnull NBTTagCompound compound) {
         super.readFromNBT(compound);
+        customName = compound.getString("CustomName");
```

Comment:
Can we wrap this around a `if (compound.hasKey("CustomName"))` for existing world compatibility?

Reply:
wdym, Neither `== null` nor `isEmpty()` is considered to have no custom name. if compound has no key will return an empty string `""`